cmd/ndm_daemonset/probe/udevprobe.go

```diff
@@ -385,10 +394,24 @@ func processUdevEvent(event udevevent.UdevEvent) controller.EventMessage {
 	action := event.GetAction()
 	klog.Infof("processing new event for (%s) action type %s", path, action)
 	deviceDetails := &blockdevice.BlockDevice{}
+	eventMessage := controller.EventMessage{}
+
+	deviceDetails.DevPath = path
+	// The change event handler discards the devices sent in the event message
+	// and only uses the dev path field to fetch the bd from the controller cache.
+	// Fill only dev path and do not process further in case of change events.
+	if action == udevevent.EventTypeChange {
+		if features.FeatureGates.IsEnabled(features.SizeChangeDetection) {
+			eventMessage.RequestedProbes = []string{udevProbeName,
+				sysfsProbeName}
+		} else {
+			eventMessage.RequestedProbes = []string{udevProbeName}
```

Comment:
This is for detecting the filesystem changes right? This also need to be done only if the feature gate is enabled.